# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1466179900 25200
#      Fri Jun 17 09:11:40 2016 -0700
# Node ID 45c5f442628981915d103f5028146d0114903dbe
# Parent  2f33052c68fffcac183744d02a452fefc4116b63
Use library wrapper for one more instance of WNOHANG (bug #48253)

* sighandlers.cc (octave_child_list::octave_child_list_rep::wait): Use
octave::sys::wnohang wrapper.

diff --git a/libinterp/corefcn/sighandlers.cc b/libinterp/corefcn/sighandlers.cc
--- a/libinterp/corefcn/sighandlers.cc
+++ b/libinterp/corefcn/sighandlers.cc
@@ -1000,17 +1000,17 @@ OCL_REP::wait (void)
       octave_child& oc = *p;
 
       pid_t pid = oc.pid;
 
       if (pid > 0)
         {
           int status;
 
-          if (octave::sys::waitpid (pid, &status, WNOHANG) > 0)
+          if (octave::sys::waitpid (pid, &status, octave::sys::wnohang ()) > 0)
             {
               oc.have_status = 1;
 
               oc.status = status;
 
               retval = true;
 
               break;
